File: c++/evenArray.cpp
#include <iostream>
using namespace std;

int main(){
	int t;
	cin >> t;
	while (t >= 1){
		int n;
		cin >> n;
		int array[n];
		int odd{}, even{};
		for (int i = 0; i < n; ++i){
			cin >> array[i];
			if (array[i] % 2 != i % 2){
				if (array[i] % 2 == 0)	++odd;
				else	++even;
			}
		}
		cout << (odd != even? -1: even) << endl;
		--t;
	}
	return 0;
}

File: python/evenArray.py
# for _ in range(int(input())):
# 	n = int(input())
# 	array = list(map(int, input().strip().split()))
# 	odd, even = 0, 0
# 	for ind, a in enumerate(array):
# 		if ind % 2 != a % 2:
# 			if a % 2 == 0:
# 				even += 1
# 			else:
# 				odd += 1
# 	print(odd if odd == even else -1)

for _ in range(int(input())):
	n = int(input())
	array = list(map(int, input().strip().split()))
	validity = [1 if (ind % 2 == x % 2)
				else 0
				for ind, x in enumerate(array)]	
	invalid = validity.count(0)
	print(invalid // 2 if invalid % 2 == 0 else -1)k